inst/scripts/insert_image.R

```diff
@@ -0,0 +1,59 @@
+local({
+  txt_input = function(..., width = '100%') shiny::textInput(..., width = width)
+
+  blogfile = rstudioapi::getActiveDocumentContext()
+  if(blogfile$path == '') stop('Please select the blog post source file before using this addin', call. = FALSE)
+
+  imgdir = file.path(
+    gsub('/content/.*', '', normalizePath(blogfile$path)),
+    'static',
+    dirname(gsub('.*content/', '', normalizePath(blogfile$path))),
+    paste0(xfun::sans_ext(basename(blogfile$path)), '_files')
+  )
+
+  shiny::runGadget(
+    miniUI::miniPage(miniUI::miniContentPanel(
+      shiny::fillRow(
+        shiny::fileInput('newimg', 'Image', placeholder = 'Select external image'),
+        height = '90px'
+      ),
+      shiny::fillRow(
+        txt_input('imgwidth', 'Image width', '', '(optional) Example: 400 for 400px'),
+        txt_input('imgheight', 'Image height', '', '(optional) Example: 200 for 200px'),
+        height = '70px'),
+      shiny::fillRow(
+        txt_input('imgalt', 'Image alternative text', '', '(optional) Example: awesome screenshot'),
+        height = '70px'
+      ),
+      miniUI::gadgetTitleBar(NULL)
+    )),
+    server = function(input, output, session) {
+      shiny::observeEvent(input$done, {
+        if (is.null(input$newimg)) return(
+          warning('You have to choose an image!', call. = FALSE)
+        )
+        dir.create(imgdir, showWarnings = FALSE)
+        file.copy(input$newimg$datapath, file.path(imgdir, input$newimg$name))
+        if(file.exists(file.path(imgdir, input$newimg$name))) message(paste('successfully copied the image to', file.path(imgdir, input$newimg$name)))
+        
+        imgsrc = paste0(
+          "/",
+          basename(dirname(imgdir)),
+          "/",
+          basename(imgdir),
+          "/",
+          input$newimg$name
+        )
+                
+        image_text = if(input$imgwidth == '' & input$imgheight == '') shiny::img(src = imgsrc) else if (input$imgwidth == '') shiny::img(src = imgsrc, height = input$imgheight) else if(input$imgheight == '') shiny::img(src = imgsrc, width = input$imgwidth) else shiny::img(src = imgsrc, width = input$imgwidth, height = input$imgheight)
```

Comment:
This is obviously way too wide. You don't have to avoid curly braces.

BTW, personally I prefer the Markdown syntax `![]()`, because HTML code is not portable (you won't be able to build the post to, say, PDF). If both width and height are empty, use Markdown.

A fancier feature would be to use the **magick** package to convert the image to the desired dimension.

Yet another feature would be to allow links on images.

Reply:
I made markdown the default when no height/width are specified.

As for resizing the image. Well, I normally preview the website and adjust image heights/widths as necessary afterwards. Resizing the image makes sense for non-html outputs (aka, when using the markdown syntax).